### PR TITLE
Electron draft packaging apps

### PR DESCRIPTION
**Pull request by brianamarie**
## Overview
**TL;DR**
This pull request is for the draft for the second course of the [Electron path](https://github.com/github/training-kit/projects/2), "packaging apps". 

_Note: This pull request is against base "electron-course", the long-running dev branch for the electron path._ 

### Questions
- Do we need to approach packaging differently based on the OS of the user? 
- Is [`electron-packager`](https://github.com/electron-userland/electron-packager) the right tool to use? 
- Do we want to recommend a tool for getting icons into the correct file format? 
- Do we want to get into removing binaries? 

### Next Steps
- ✏️ @github/services-training write and draft content 
- 💻 @github/services-training to test content to ensure it works 
- 💭 @lee-dohm to advise on questions above ☝️ 

### Review
To test and review, I recommend first testing the [getting started PR](https://github.com/github/training-kit/pull/538), then going through this material one of two ways: 

- Do the steps as if you're trying to complete the activity once it's spun up
- Go through the files and fill out/edit the content for readability or added resources

### CODE REVIEW

**Comment by brianamarie** [comment]
@github/services-training @lee-dohm I think this small course is also ready for some 👀. To test and review, I recommend first testing the [getting started PR](https://github.com/github/training-kit/pull/538), then going through this material one of two ways: 

- Do the steps as if you're trying to complete the activity once it's spun up
- Go through the files and fill out/edit the content for readability or added resources

File: paths/electron/packaging-apps/02-using-electron-packager.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+facilitator: false
+sidebar:
+  nav: "electron-packaging-apps"
+main-content: |
+
+  ## Using `electron-packager`
+
+  Now that we have installed `electron-packager`, we're ready to use it. We could type out the entire command every time, but it's best practice to add a build script to the `package.json` file. This means that every time we run the default build script, our specific list of commands will be run.
+
+  1. In the `package.json` file, add a build script `"build": "electron-packager . app-name --ignore=node_modules/electron-*"` in "scripts" of package.json. (Ignore all dependency apps the same way.)
```
Comment (hectorsector) [inline paths/electron/packaging-apps/02-using-electron-packager.md]: For users that are newer to `package.json` files (and I think it'll be good practice in general), we should provide links to examples of what the file should look like at each point. This should be pretty easy for us to provide (we can just give links to commits in a repo, for example).

File: paths/electron/packaging-apps/02-using-electron-packager.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  1. Run the new build script with `npm run build`
```
Comment (hectorsector) [inline paths/electron/packaging-apps/02-using-electron-packager.md]: This is probably due to whatever errors I had back in the getting started course, but I get an error:

screen shot 2017-08-04 at 5 36 47 pm

File: paths/electron/packaging-apps/03-adding-an-icon.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  To convert the image to the appropriate format, we recommend [this tool](TBD).
+
+  ### Including the Image in the Build
+  1. In your project's root directory, create a new directory called `Icon`: `mkdir Icon`
+  1. Add all of the image files to the `Icon` directory
+  1. Replace icon in Contents>Resources directory in build script:
+
+                  ```json
+                  "build": "electron-packager . app-name && cp Icon.icns Path/To/New/Icon"
```
Comment (hectorsector) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: This command didn't work for me at all. I think it's supposed to be something along these lines, but it didn't work either:

```json
"build": "electron-packager . app-name && cp Icon/icon.png app-name-darwin-x64/app-name.app/Contents/Resources/Icon.icns"
```

File: paths/electron/packaging-apps/03-adding-an-icon.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                  ```
+                  
+  1. Delete app: `rm -rf AppName.app`
```
Comment (hectorsector) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: This is inside a folder named after the OS: `app-name-darwin-x64` for me.

Comment (hectorsector) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: I'm guessing these build artifacts shouldn't be committed to Git, so should we include that in the `.gitignore`, or at least tell them?

File: paths/electron/packaging-apps/04-lock-source-code.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  nav: "electron-packaging-apps"
+main-content: |
+
+  ## Lock Source Code
+
+  When people download the application, you may or may not want them to have direct access to the source code. With a node package called [`asar`](https://www.npmjs.com/package/asar), you can control this. If you're not concerned with this, you can skip this step.
+
+  1. Archive capability with asar
+  1. `npm install asar --save-dev`
+  1. Create another script in package.json called "package": "asar pack AppName.app/Contents/Resources/app MyApp.app/Contents/Resources/app.assar"
```
Comment (hectorsector) [inline paths/electron/packaging-apps/04-lock-source-code.md]: How can I tell if this worked?

File: paths/electron/packaging-apps/02-using-electron-packager.md (hunk 1 above)
Comment (brianamarie) [inline paths/electron/packaging-apps/02-using-electron-packager.md]: There's no 👏 reaction emoji but I agree completely. Committing a todo now so we don't forget.

File: paths/electron/packaging-apps/03-adding-an-icon.md (hunk 3 above)
Comment (brianamarie) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: Gotcha. I think for this one to work, we need to have the image part fleshed out. Did you already create the icon file?

File: paths/electron/packaging-apps/03-adding-an-icon.md (hunk 4 above)
Comment (brianamarie) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: Ooh that's a great point. adding that as well to the todo

File: paths/electron/packaging-apps/04-lock-source-code.md (hunk 5 above)
Comment (brianamarie) [inline paths/electron/packaging-apps/04-lock-source-code.md]: That's a great question that I don't have an answer for. 🤔 I guess when someone downloads the tarball or whatever compiled app, they shouldn't see the other source code files?

File: paths/electron/packaging-apps/03-adding-an-icon.md (hunk 3 above)
Comment (hectorsector) [inline paths/electron/packaging-apps/03-adding-an-icon.md]: I did.

File: paths/electron/packaging-apps/02-using-electron-packager.md (hunk 1 above)
Comment (hollenberry) [inline paths/electron/packaging-apps/02-using-electron-packager.md]: 👍 . Do you think it might be worth clarifying the difference between a package.json and a package-lock.json file as well?

Comment (brianamarie) [inline paths/electron/packaging-apps/02-using-electron-packager.md]: Good question...I would need to review if the boilerplate creates a package-lock file. I think if the boilerplate makes one, then yes. But, if it doesn't, I think we should leave it off and let that fall under the "other Javascript" that will need to be learned outside of this course. How does that sound?

**Comment by brianamarie** [comment]
Merging drafts so we can start a clean, more formal review. 🚢
